DM-27282: make is_valid=True the default for find_messages

r-owen opened the pull request.

afausti reviewed (approved):
That makes `find_messages` should return only valid messages by default. Note that tox is failing running the tests and in checking the typing annotations though.

r-owen commented:
Thank you for reporting that tox was failing the type annotation test (I should have checked that). I fixed that. I found and implemented another needed update in test_edit_message as well.

I think fixing the github test will require fixing a known issue (with a ticket) installing daf_butler. Locally `tox` now succeeds.
